Tasks/Common/dotNetCliWrapper.ts

```diff
@@ -112,27 +201,76 @@ export class DotNetCliWrapper {
         args.push('--disable-interactive')
         args.push('true')
 
-        return this.executeAsync(args, additionalArgs)
+        return await this.execute(args, additionalArgs)
     }
 
-    public async executeAsync(args: string[], additionalArgs: string): Promise<void> {
-        const dotnetPath = tl.which('dotnet', true)
-        console.log('Path to tool: ' + dotnetPath)
+    private getLambdaToolPath(): string {
+        const dotnetPath = path.join('~', '.dotnet', 'tools')
+
+        let lambdaPath = ''
+        try {
+            lambdaPath = tl.which(this.lambdaDotnetTool, true)
+
+            return lambdaPath
+        } catch (e) {}
+        lambdaPath = path.join(dotnetPath, `${this.lambdaDotnetTool}.exe`)
+        try {
+            fs.accessSync(lambdaPath)
 
-        const dotnet = tl.tool(dotnetPath)
+            return lambdaPath
+        } catch (e) {}
+        lambdaPath = path.join(dotnetPath, this.lambdaDotnetTool)
+        try {
+            fs.accessSync(lambdaPath)
+
+            return lambdaPath
+        } catch (e) {}
+
+        tl.debug(tl.loc('LambdaToolsPathNotFound'))
+
+        return ''
+    }
+
+    private async execute(args: string[], additionalArgs: string, additionalCommandLineOptions?: any): Promise<number> {
+        const toolPath = this.getLambdaToolPath()
+        let dotnet: ToolRunner
+        if (toolPath) {
```

Comment:
I would suggest some logging statements to indicate which path was taken here.

Reply:
There is a statement above: tl.debug(tl.loc('LambdaToolsPathNotFound'))